features/device-config/rest/src/main/java/org/opennms/features/deviceconfig/rest/impl/DefaultDeviceConfigRestService.java

```diff
@@ -235,9 +236,42 @@ public Response getDeviceConfigsByInterface(Integer ipInterfaceId, String config
 
     /** {@inheritDoc} */
     @Override
-    public void deleteDeviceConfig(long id) {
-        throw new UnsupportedOperationException("Delete device config not yet supported.");
-        // deviceConfigDao.delete(id);
+    public Response deleteDeviceConfigs(List<Long> ids) {
+        if (CollectionUtils.isEmpty(ids)) {
+            LOG.debug("Bad request : empty or null DeviceConfig Id");
+            return Response.status(Status.BAD_REQUEST).entity("Invalid 'id' parameter").build();
+        }
+        sessionUtils.withTransaction(() -> {
+            try {
+                final List<DeviceConfig> deviceConfigList = ids.stream()
+                        .map(deviceConfigDao::get)
+                        .map(DeviceConfig.class::cast)
+                        .collect(Collectors.toList());
+                deviceConfigDao.deleteDeviceConfigs(deviceConfigList);
+            } catch (Exception e) {
+                LOG.error("Exception while deleting device configs, one or more ids not valid {}", e);
+            }
+        });
+
+        return Response.noContent().build();
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public Response deleteDeviceConfig(long id) {
+       sessionUtils.withTransaction(() -> {
+            try {
+                final DeviceConfig dc = deviceConfigDao.get(id);
+                if (dc == null) {
+                    LOG.debug("could not find device config data for id: {}", id);
```

Comment:
This should return a 404 if not found

Reply:
code changed to return 404 (transaction code also changed since earlier implemention was returning null only)